core/sessions/localauth/reaper_test.go

```diff
@@ -53,32 +55,34 @@ func TestSessionReaper_ReapSessions(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			user := cltest.MustRandomUser(t)
-			require.NoError(t, orm.CreateUser(&user))
-
-			session := sessions.NewSession()
-			session.Email = user.Email
-
-			_, err := db.Exec("INSERT INTO sessions (last_used, email, id, created_at) VALUES ($1, $2, $3, now())", test.lastUsed, user.Email, test.name)
-			require.NoError(t, err)
-
 			t.Cleanup(func() {
-				_, err2 := db.Exec("DELETE FROM sessions where email = $1", user.Email)
-				require.NoError(t, err2)
+				clearSessions(t, db.DB)
 			})
 
+			_, err := db.Exec("INSERT INTO sessions (last_used, email, id, created_at) VALUES ($1, $2, $3, now())", test.lastUsed, cltest.APIEmailAdmin, test.name)
+			require.NoError(t, err)
+
 			r.WakeUp()
-			<-r.(interface {
-				WorkDone() <-chan struct{}
-			}).WorkDone()
-			sessions, err := orm.Sessions(0, 10)
-			assert.NoError(t, err)
 
 			if test.wantReap {
-				assert.Len(t, sessions, 0)
+				gomega.NewWithT(t).Eventually(func() []sessions.Session {
+					sessions, err := orm.Sessions(0, 10)
+					assert.NoError(t, err)
+					return sessions
+				}).Should(gomega.HaveLen(0))
 			} else {
-				assert.Len(t, sessions, 1)
+				gomega.NewWithT(t).Consistently(func() []sessions.Session {
+					sessions, err := orm.Sessions(0, 10)
+					assert.NoError(t, err)
+					return sessions
+				}).Should(gomega.HaveLen(1))
 			}
 		})
 	}
 }
+
+// clearSessions removes all sessions.
+func clearSessions(t *testing.T, db *sql.DB) {
```

Comment:
See my comment above: all uses of this should be targeted at deleting specific rows.

Reply:
Done!